src/FormField/AutoComplete.php

```diff
@@ -157,4 +156,59 @@ public function getInput()
             'value' => $this->getValue(),
         ]);
     }
+
+    /**
+     * Set Semantic-ui dropdown module settings.
+     *
+     * For example, using this setting will automatically submit
+     * form when field value is changes.
+     * $ac->setDropdownSettings(['allowReselection' => true,
+     *                           'selectOnKeydown' => false,
+     *                           'onChange'        => new atk4\ui\jsExpression('function(value,t,c){
+     *                                                          if ($(this).data("value") !== value) {
+     *                                                            $(this).parents(".form").form("submit");
+     *                                                            $(this).data("value", value);
+     *                                                          }
+     *                                                         }'),
+     *                          ]);
+     *
+     * @param array $settings
+     *
+     * @return $this
+     */
+    public function setDropdownSettings($settings)
+    {
+        $this->dropdownSettings = $settings;
+
+        return $this;
+    }
+
+    /**
+     * Set Semantic-ui Api settings to use with dropdown.
+     *
+     * @param array $config
+     *
+     * @return $this
+     */
+    public function setApiConfig($config)
+    {
+        $this->apiConfig = array_merge($this->apiConfig, $config);
+
+        return $this;
+    }
+
+    public function renderView()
+    {
+        $chain = new jQuery('#'.$this->name.'-ac');
+
+        $settings = array_merge([
+            'fields'      => ['name' => 'name', 'value' => 'id'/*, 'text' => 'description'*/],
+            'apiSettings' => array_merge($this->apiConfig, ['url' => $this->getCallbackURL().'&q={query}']),
+        ], $this->dropdownSettings);
```

Comment:
my proposal was to move this line into a function and call a function, so if user would want to do something differently he could technically override anything. This way it's still not so flexible as you can't modify some things. 

Also overal implementation became more complex.

Reply:
You should be able to override everything using array_merge no? Not sure to exactly understand what you mean by adding a method and how to render the call to $('#ac_id').dropdown([]) using such method, other then in renderView. Can you provide a sample?

Reply:
well for example i can't omit 'apiSettings'. I can probably set it to false. OOP-approach is more flexible.